CRM/Member/Form/Membership.php

```diff
@@ -182,7 +182,7 @@ public function preProcess() {
       $contributionID = CRM_Member_BAO_Membership::getMembershipContributionId($this->_id);
       // check delete permission for contribution
       if ($this->_id && $contributionID && !CRM_Core_Permission::checkActionPermission('CiviContribute', $this->_action)) {
-        CRM_Core_Error::fatal(ts("This Membership is linked to a contribution. You must have 'delete in CiviContribute' permission in order to delete this record."));
+        CRM_Core_Error::statusBounce(ts("This Membership is linked to a contribution. You must have 'delete in CiviContribute' permission in order to delete this record."));
```

Comment:
This should have it's own PR but makes sense - as long as you have tested it (which I assume you did when creating the change) I'm happy to merge a PR with just this change